### ISSUE TITLE
Emoji's in string output

### ISSUE THREAD

**Issue by benmccallum** [ChilliCream/hotchocolate#2301]
**Describe the bug**
@anthony-keller and @maraisr noticed emojis come out as jibberish. 

**To Reproduce**
Return an emoji from a string field.

**Expected behavior**
Comes out correctly.

**Desktop (please complete the following information):**
 - OS: Win 10 latest
 - Version: v10.5.2

**Comment by benmccallum** [comment]
If you guys have got any more to add, feel free to add Anth/Marais. Just getting this logged.

**Comment by anthony-keller** [comment]
E.g.

> "Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience������"

should be

> Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience👍

**Comment by michaelstaib** [comment]
I suspect this bug with how we serialize the string with System.Text.Json looks like a simple encoding issue :) always these encoding things :D

**Comment by maraisr** [comment]
Just strange, coz emoji's are in the standard unicode space. So what encoding is it then?

**Comment by MariaCobretti** [comment]
🤑 emojis are working for me .. used the banana cake pop client

**Comment by michaelstaib** [comment]
@benmccallum when does this problem occur? Are the strings broken when you write them to the server (input) or when you query them (output).

Also if they are broken on input are you sending them in through variables or as GraphQL literals in the query?

**Comment by benmccallum** [comment]
I honestly haven't looked at this one. @anthony-keller and @maraisr can chime in. I know it's when querying, but I'm not sure how it's stored, rendered (playground or into html) etc.

**Comment by anthony-keller** [comment]
The string is stored in SQL in a column with these properties:
![image](https://user-images.githubusercontent.com/31554139/93394555-c1a8f300-f8b7-11ea-9287-9500bdac626a.png)

It renders like this in SSMS
![image](https://user-images.githubusercontent.com/31554139/93394592-d08fa580-f8b7-11ea-846c-f038488eda99.png)

We query using EF and project it to a DTO using AutoMapper.

We have a REST endpoint available as well that can return reviews. It returns the following response.

`
   "customerRatings": [
        {
            "ratingID": 12796,
            "comments": "Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience👍",
        }
    ],
 `

**Comment by anthony-keller** [comment]
If I execute this query:

`
query {
  supplier (id:194) {
    __typename
    ...on Supplier {
      rating {
        reviews {
          edges {
            node {
              comment
            }
          }
        }
      }
    }
  }
}
`
I get this response
`
{
  "data": {
    "supplier": {
      "__typename": "Supplier",
      "rating": {
        "reviews": {
          "edges": [
            {
              "node": {
                "comment": "Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience������"
              }
            },
            {
              "node": {
                "comment": "During this normal servicing, they have genuinely identified the problem associated with my car. I have to go there after 5k, to replace front brakes shoe."
              }
            },
            {
              "node": {
                "comment": null
              }
            },
            {
              "node": {
                "comment": null
              }
            },
            {
              "node": {
                "comment": "Front break shoes replacement was done perfectly and cost effectively. I loved the service provided, and would like to visit again for my scheduled servicing."
              }
            },
            {
              "node": {
                "comment": null
              }
            },
            {
              "node": {
                "comment": "I didn’t had to tell them what to do, They knew what was to be done. Perfect servicing experience , I am fully satisfied."
              }
            },
            {
              "node": {
                "comment": null
              }
            },
            {
              "node": {
                "comment": null
              }
            },
            {
              "node": {
                "comment": null
              }
            }
          ]
        }
      }
    }
  }
}
`

**Comment by anthony-keller** [comment]
`var comment = Ratings.Where (r => r.RatingID == 12796).Select(r => r.Comments).Single();

comment.Dump();

var asciiBytes = Encoding.ASCII.GetBytes(comment);

var asciiString = Encoding.ASCII.GetString(asciiBytes);

asciiString.Dump();`

`Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience👍
Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience??

`

**Comment by anthony-keller** [comment]
Running a query directly against the stitched service

```
query {
  ratingForSupplierDbId(supplierDbId: 194, isEligibleForHighlyRecommended:false) {
    reviews {
      nodes {
        comment
      }
    }
  }
}
```
Gets a valid response
```
{
  "data": {
    "ratingForSupplierDbId": {
      "reviews": {
        "nodes": [
          {
            "comment": "Honest, friendly, customer-focused. Willing to go that little bit further to provide a positive experience👍"
          },
          {
            "comment": "During this normal servicing, they have genuinely identified the problem associated with my car. I have to go there after 5k, to replace front brakes shoe."
          },
          {
            "comment": null
          },
          {
            "comment": null
          },
          {
            "comment": "Front break shoes replacement was done perfectly and cost effectively. I loved the service provided, and would like to visit again for my scheduled servicing."
          },
          {
            "comment": null
          },
          {
            "comment": "I didn’t had to tell them what to do, They knew what was to be done. Perfect servicing experience , I am fully satisfied."
          },
          {
            "comment": null
          },
          {
            "comment": null
          },
          {
            "comment": null
          }
        ]
      }
    }
  }
}
```

**Comment by anthony-keller** [comment]
It would appear to be caused by something in the stitching process. Hope this helps track it down.

**Comment by michaelstaib** [comment]
Yes it helps. I tried it with a local GraphQL server and could not get the issue. Will have a look at the stitching layer.

**Comment by michaelstaib** [comment]
This one is solved with version 11.

**Comment by benmccallum** [comment]
Hey @michaelstaib , this still seems to be a problem on `11.0.8`. 

Works when queried direct from downstream service:
![image](https://user-images.githubusercontent.com/157336/105014536-d998ab00-5a40-11eb-816b-03ac21056a72.png)

But fails when via gateway:
![image](https://user-images.githubusercontent.com/157336/105014581-e61d0380-5a40-11eb-94a8-1b37d26095aa.png)

(using BCP)

**Comment by michaelstaib** [comment]
We moved this to the august release since the august release will focus on stitching.

**Comment by sblomstrand** [comment]
Any update on this? I'm having a similar issue when saving data through the gateway but works when i go directly to the stitched server.

**Comment by sblomstrand** [comment]
An ugly test suggesting it may have something to do with System.Text.Json using `.AddHttpMessageHandler` and rewriting the request and response using Newtonsoft.Json actually results in correct parsing.
`variables`:
```
{
  "text": "😬"
}
```

```
    services.AddHttpClient("schemaName", (sp, client) =>
            {
                client.BaseAddress = new Uri("url");
            }).AddHttpMessageHandler<RewriteRequestResponseHandler>();
```

```
    public class RewriteRequestResponseHandler : DelegatingHandler
    {
        protected override async Task<HttpResponseMessage> SendAsync(
            HttpRequestMessage request,
            CancellationToken cancellationToken)
        {
            if (request.Content != null)
            {
                var stringContent = await request.Content.ReadAsStringAsync(); // "text": "\uD83D\uDE2C"
                var json = Newtonsoft.Json.JsonConvert.DeserializeObject<dynamic>(stringContent).ToString(); // "text": "😬"
                request.Content = new StringContent(json, encoding: System.Text.Encoding.UTF8, "application/json");
            }

            var result = await base.SendAsync(request, cancellationToken);

            if (result.Content != null)
            {
                var resultStringContent = await result.Content.ReadAsStringAsync(); // "text":"\uD83D\uDE2C"
                var json = Newtonsoft.Json.JsonConvert.DeserializeObject<dynamic>(resultStringContent).ToString(); // "text": "😬"
                result.Content = new StringContent(json, encoding: System.Text.Encoding.UTF8, "application/json");
            }

            return result;
        }
    }
```

I tried the same thing using `System.Text.Json.JsonSerialiser` with `JsonSerializerOptions`:
```
        JsonSerializerOptions
        {
            Encoder = System.Text.Encodings.Web.JavaScriptEncoder.Create(System.Text.Unicode.UnicodeRanges.All)
        }
```
but that did not work

**Comment by sblomstrand** [comment]
Don't close, still needs to be fixed.

**Comment by bjharo** [comment]
I'm seeing this as well using Strawberry Shake as a client against a Hot Chocolate server. Can confirm that @sblomstrand test results in the correct parsing.

**Comment by michaelstaib** [comment]
HotChocolate.Stitching is now in legacy mode and is replaced by HotChocolate.Fusion. I am closing this issue as we essentially froze the fusion code.

You can join the preview of Hot Chocolate Fusion now on Slack

**Comment by rickardp** [comment]
This is possibly not a bug in the stitching per se, but in the request parser. It will be apparent in all stiching since System.Text.Json is used (which escapes output). Please see the issue #6488 I created.
